src/ol_infrastructure/infrastructure/aws/opensearch/__main__.py

```diff
@@ -39,27 +39,65 @@
 cluster_instance_type = search_config.get("instance_type") or "t3.medium.elasticsearch"
 disk_size = search_config.get_int("disk_size_gb") or 30  # noqa: WPS432
 
-search_security_group = aws.ec2.SecurityGroup(
-    "opensearch-security-group",
-    name_prefix=f"{environment_name}-opensearch-",
-    tags=aws_config.merged_tags({"Name": f"{environment_name}-opensearch"}),
-    description="Grant access to the OpenSearch service domain",
-    egress=default_egress_args,
-    ingress=[
+##########
+# CREATE #
+##########
+
+# Networking
+if search_config.get_bool("public_web"):
+    sg_ingress_rules = [
+        aws.ec2.SecurityGroupIngressArgs(
+            from_port=DEFAULT_HTTPS_PORT,
+            to_port=DEFAULT_HTTPS_PORT,
+            cidr_blocks=["0.0.0.0/0"],
+            protocol="tcp",
+        )
+    ]
+else:
+    sg_ingress_rules = [
         aws.ec2.SecurityGroupIngressArgs(
             from_port=DEFAULT_HTTPS_PORT,
             to_port=DEFAULT_HTTPS_PORT,
             cidr_blocks=[target_vpc["cidr"]],
             protocol="tcp",
         )
-    ],
+    ]
+search_security_group = aws.ec2.SecurityGroup(
+    "opensearch-security-group",
+    name_prefix=f"{environment_name}-opensearch-",
+    tags=aws_config.merged_tags({"Name": f"{environment_name}-opensearch"}),
+    description="Grant access to the OpenSearch service domain",
+    egress=default_egress_args,
+    ingress=sg_ingress_rules,
     vpc_id=target_vpc["id"],
 )
 
+# OpenSearch Domain
+conditional_kwargs = {}
+if not search_config.get_bool("public_web"):
+    conditional_kwargs["vpc_options"] = aws.elasticsearch.DomainVpcOptionsArgs(
+        subnet_ids=target_vpc["subnet_ids"][:3],
+        security_group_ids=[search_security_group.id],
+    )
+
+if search_config.get_bool("secured_cluster"):
+    ## TODO 20220307 MAD, look at encryption @ rest and other options
+    conditional_kwargs[
+        "domain_endpoint_options"
+    ] = aws.elasticsearch.DomainDomainEndpointOptionsArgs(
+        enforce_https=True,
+        tls_security_policy="Policy-Min-TLS-1-2-2019-07",
+    )
+    conditional_kwargs[
+        "node_to_node_encryption"
+    ] = aws.elasticsearch.DomainNodeToNodeEncryptionArgs(
+        enabled=True,
+    )
+
 search_domain = aws.elasticsearch.Domain(
     "opensearch-domain-cluster",
     domain_name=f"opensearch-{environment_name}"[:SEARCH_DOMAIN_NAME_MAX_LENGTH],
-    elasticsearch_version=search_config.get("engine_version") or "7.10",
+    elasticsearch_version=search_config.get("engine_version") or 7.10,
```

Comment:
Will this work as a decimal value, or does it need to remain as a string?

Reply:
oof good catch.